tomopy/prep/alignment.py

```diff
@@ -443,3 +455,240 @@ def shift_images(prj, sx, sy):
+        # assuming a rectangle type slit box 
+        col_std = medfilt(np.std(img, axis=0), kernel_size=medfilt_kernel_size)
+        row_std = medfilt(np.std(img, axis=1), kernel_size=medfilt_kernel_size)
+        # NOTE: in the tiff img
+        #  x is col index, y is the row index  ==> key point here !!!
+        #  img slicing is doen with img[row_idx, col_idx]
+        #  ==> so the image idx and corner position are FLIPPED!
+        _left   = np.argmax(np.gradient(col_std))
+        _right  = np.argmin(np.gradient(col_std))
+        _top    = np.argmax(np.gradient(row_std))
+        _bottom = np.argmin(np.gradient(row_std))
+
+        cnrs = np.array([[_left , _top    ],
+                         [_left , _bottom ],
+                         [_right, _bottom ],
+                         [_right, _top    ],
+                        ])
+    else:
+        # predefine all quadrants
+        # Here let's assume that the four corners of the slit box are in the 
+        # four quadrant defined by the center of the image
+        # i.e.
+        #  uppper left quadrant: img[0     :cnt[1], 0     :cnt[0]]  => quadarnt origin =  (0,           0)
+        #  lower  left quadrant: img[cnt[1]:      , 0     :cnt[0]]  => quadarnt origin =  (cnt[0],      0)
+        #  lower right quadrant: img[cnt[1]:      , cnt[0]:      ]  => quadarnt origin =  (cnt[0], cnt[1])
+        #  upper right quadrant: img[0     :cnt[1], cnt[0]:      ]  => quadarnt origin =  (0,      cnt[1])
+        cnt = [int(cols/2), int(rows/2)]  # center of image that defines FOUR quadrants
+        Quadrant  = namedtuple('Quadrant', 'img col_func, row_func')
+        quadrants = [Quadrant(img=img[0:cnt[1], 0:cnt[0]], col_func=np.argmax, row_func=np.argmax),  # upper left,  1st quadrant
+                     Quadrant(img=img[cnt[1]: , 0:cnt[0]], col_func=np.argmax, row_func=np.argmin),  # lower left,  2nd quadrant
+                     Quadrant(img=img[cnt[1]: , cnt[0]: ], col_func=np.argmin, row_func=np.argmin),  # lower right, 3rd quadrant
+                     Quadrant(img=img[0:cnt[0], cnt[1]: ], col_func=np.argmin, row_func=np.argmax),  # upper right, 4th quadrant
+                    ]
+        # the origin in each quadrants ==> easier to set it here
+        quadrantOrigins = np.array([[0,           0],  # upper left,  1st quadrant
+                                    [0,      cnt[1]],  # lower left,  2nd quadrant
+                                    [cnt[0], cnt[1]],  # lower right, 3rd quadrant
+                                    [cnt[1],      0],  # upper right, 4th quadrant
+                                   ])
+        # init four corners
+        cnrs = np.zeros((4, 2))
+        if method.lower() == 'quadrant':
+            # the standard quadrant method
+            for i, q in enumerate(quadrants):
+                cnrs[i,:] = np.array([q.col_func(np.gradient(medfilt(np.std(q.img, axis=0), kernel_size=medfilt_kernel_size))),  # x is col_idx

```

Comment:
Long line?

Reply:
autopep8 fix this problem by breaking it into multiple lines.